test/OpenTelemetry.Tests/Logs/LogEmitterTests.cs

```diff
@@ -173,5 +173,37 @@ public void LogEmitterUnspecifiedTimestampTest()
             Assert.Equal(timestamp, logRecord.Timestamp);
             Assert.Equal(DateTimeKind.Unspecified, logRecord.Timestamp.Kind);
         }
+
+        [Fact]
+        public void CircularLogTest()
+        {
+            var options = new OpenTelemetryLoggerOptions();
+
+            using var provider1 = new OpenTelemetryLoggerProvider(options);
+
+            var emitter1 = provider1.CreateEmitter();
+            var emitter2 = provider1.CreateEmitter();
+
+            provider1.AddProcessor(new SimpleLogRecordExportProcessor(new NoopExporter()
+            {
+                Emitter = emitter1,
+            }));
+
+            emitter2.Emit(default);
+        }
+
+        private sealed class NoopExporter : BaseExporter<LogRecord>
+        {
+            public LogEmitter Emitter { get; set; }
+
+            public override ExportResult Export(in Batch<LogRecord> batch)
+            {
+                var result = ExportResult.Success;
+
+                this.Emitter?.Emit(default);
```

Comment:
> I'm thinking a special ThreadStatic might be cheaper to detect re-entrant logs and bail out in that case.

A simple solution using `ThreadStatic` might work just fine in this case, but if this line were instead:

```csharp
Task.Run(() => this.Emitter?.Emit(default));
```

... you'd have problems.